lib/blueprinter/view_collection.rb

```diff
@@ -58,12 +58,27 @@ def sortable_fields(view_name)
       fields
     end
 
-    def merge_fields(source_fields, included_fields)
-      if sort_by_definition
-        included_fields.merge(source_fields)
+    # select and order members of fields according to traversal of the definition_orders
+    def sort_by_def(view_name, fields)
+      ordered_fields = {}
+      views[:default].definition_order.each { |definition| add_to_ordered_fields(ordered_fields, definition, fields, view_name)  }
+      ordered_fields.values
+    end
+
+    # view_name_filter allows to follow definition order all the way down starting from the view_name given to sort_by_def()
+    # but include no others at the top-level
+    def add_to_ordered_fields(ordered_fields, definition, fields, view_name_filter = nil)
+      if definition.view?
+        if view_name_filter.nil? || view_name_filter == definition.name
+          views[definition.name].definition_order.each { |_definition| add_to_ordered_fields(ordered_fields, _definition, fields) }
+        end
       else
-        source_fields.merge(included_fields)
+        ordered_fields[definition.name] = fields[definition.name]
       end
     end
+
+    def merge_fields(source_fields, included_fields)
+      source_fields.merge! included_fields
```

Comment:
Any reason we wanted to make this `merge!` vs `merge`? If we stick with `merge!` we should probably change this method name to `merge_fields!` and there is no longer a need to do: `fields = merge_fields!(fields...` but I think I'd prefer this method without side-effects.

Reply:
This comment is notable. I think we can refactor it after, for now I think we can go ahead and merge this